### PR TITLE
Issue 531, fixing weird sticky date behavior

### PR DESCRIPTION
**Pull request by asharonbaltazar**
Dates in the Activities page now scroll as expected. 

Something needs to be done about the filter when the viewport is of mobile width. See:

image

### CODE REVIEW

**Review by csell5 — CHANGES_REQUESTED** [review]
FYI, I did update the branch to rebase on top of your other changes.

There are a few changes needed here.

First things are now cutting off the top of the items in the grid
image

I think the spirt of the issue wasn't so much about the placement of the header, but it was the placement of the header when it was the last item for that slot. thus you end up like this:

image

It would be a cleaner approach if that last item in that slot could put the header out of the way vs, the next header. Clearly not designed that way, and this isn't a super critical issue by any means.  

The problem I see with your changes.
* the time is is never vertically aligned which is hard to track
* Mobile layout is different than desktop so it's confusing.
* The problem with where the time is at, when a slot is "full" we're three items wide, which means that time is now fighting in the middle item.
* and in the end we still didn't solve the true issue

image

Ironically you could make an argument that in desktop mode, it just just pop out to the left alongside that main content grid. I never looked into that.

**Comment by asharonbaltazar** [comment]
I must've misunderstood the issue then. 

I'll take a look later today.

**Comment by csell5** [comment]
I know all the issues don't have the best documentation around them, so feel to ask any leading questions and I'm happy to add as much context/direction as possible. We can always setup a activity on that.us too

**Comment by asharonbaltazar** [comment]
Yeah, thanks for clarification. I don't want to seem like I didn't understand anything.

**Comment by csell5** [comment]
oh I didn't take it that way. I'm grateful for the help and anything I can do to help just let me know.

**Comment by asharonbaltazar** [comment]
image

What can we do about this? This isn't a new bug; it's been always there. Seems like the `z-index` needs to be upped on the chat bubble.

**Comment by csell5** [comment]
ok, a little back story on some tech debt here. 

We had reset the z-index of the chat bubble as it was conflicting with the warning and error boxes that pop up at the bottom. So we pushed it down.

https://github.com/ThatConference/that.us/blob/master/src/App.svelte#L35-L39

those modals could be rethought a bit but a quick fix for now I think is to just reworking the z-index of those three items so they layer right. 

FWIW The only reason there is a background on that header, was to hide the cards as they scrolled under.

**Comment by asharonbaltazar** [comment]
Other than the `z-index`, is everything okay?

**Comment by csell5** [comment]
yea but can we make one last change? I know why you removed the padding at the top so you couldn't see the gap, that's cool.

Can we add some spacing to get the top of the date to align with the top of the search box?

image

**Comment by asharonbaltazar** [comment]
Seems like it's not just the dates...

image

**Comment by csell5** [comment]
Looks like the sticky header isn't working anymore.

**Comment by asharonbaltazar** [comment]
I'm sorry, but what header? Everything seems to be working for me unless overlooked something...

**Comment by csell5** [comment]
After looking further. It appears to just be broken in Safari. I know you're on windows so maybe we just file a new issue related to just that and link it back to this pr.

**Comment by asharonbaltazar** [comment]
What should be done regarding this?

**Comment by csell5** [comment]
I'm going to take a closer look here today and see what I can suss out. might be something simple we can add into this.

File: src/components/activities/List.svelte [hunk 1]
```diff
@@ -174,40 +174,44 @@
   {#each sorted as day, d}
     <div in:fade="{{ delay: d * 200 }}">
       <h2
-        class="sticky top-2 bg-white text-3xl leading-9 font-extrabold
-        tracking-tight text-thatBlue-800 sm:text-4xl sm:leading-10 pb-2 z-10"
+        class="sticky top-0 z-1 bg-white text-3xl leading-9 font-extrabold
+        tracking-tight text-thatBlue-800 sm:text-4xl sm:leading-10 pt-4 pb-10 inline-block"
```
Comment (csell5) [inline src/components/activities/List.svelte:178]: inline-block is causing the issues in safari with sticky not sticking

File: src/components/activities/List.svelte [hunk 2]
```diff
@@ -174,40 +174,44 @@
...
       </h2>
 
       {#each day.timeSlots as ts, t}
-        <div in:fade="{{ delay: t * 500 }}" class="pb-12">
+        <div>
           <h2
-            class="sticky top-12 bg-white text-3xl leading-9 font-extrabold
-            tracking-tight text-thatOrange-400 sm:text-4xl sm:leading-10 z-10"
+            class="sticky top-14 z-1 bg-white text-3xl leading-9 font-extrabold
+            tracking-tight text-thatOrange-400 sm:text-4xl sm:leading-10 inline-block"
```
Comment (csell5) [inline src/components/activities/List.svelte:187]: inline-block is causing the issues in safari with sticky not sticking

**Review by csell5 — CHANGES_REQUESTED** [review]
I'm not sure why `inline-block` is causing sticky to not work in Safari but removing it fixes that problem. It's worth looking into a bit more.

**Comment by asharonbaltazar** [comment]
Please tell me what's happening on your end.

[Stackoverflow issue](https://stackoverflow.com/questions/38955362/displayinline-block-not-working-on-safari)

**Comment by csell5** [comment]
Kapture 2021-01-29 at 13 29 43

**Comment by asharonbaltazar** [comment]
Oh, I meant more like what the dev tools are showing. I linked an issue from StackOverflow. Apparently, it's not just us.

**Comment by asharonbaltazar** [comment]
Also noticed that when there is more than one event, the issue crops up back again. 

image

**Comment by csell5** [comment]
I'm not really sure why it doesn't work in Safari. They look exactly the same. Brave left, Safari Right.

image

**Comment by asharonbaltazar** [comment]
Can you try adding `position: -webkit-sticky;`, please?

**Comment by csell5** [comment]
that didn't work. best I can tell. the inline-block is throwing off the height for sticky, so it doesn't know where to stick. Just a guess at the moment.

**Comment by asharonbaltazar** [comment]
It might be a little unorthodox but simply explained, the `div` and header blocks have `block` applied to them by default. So if we change that `div` to something like `span`, the problem ceases to exist.

**Comment by csell5** [comment]
but you removed the inline-block also.

**Comment by asharonbaltazar** [comment]
I eliminated the need for it. Does it work for you?

**Comment by csell5** [comment]
ah, I'm following. I gotcha. yea we can roll with it.

**Comment by csell5** [comment]
fixes #531

**Comment by csell5** [comment]
I swear to you, I had just tested this in safari and it was working.. Now after the deployment it's not and it's not here either.

**Comment by asharonbaltazar** [comment]
What in the world? Are the changes live?

**Comment by csell5** [comment]
yea. when we merge into the mainline it deploys to production. It must be the monday crazy.. Let's just create a new issue for it and then link it back to this one.

**Comment by asharonbaltazar** [comment]
Also mobile needs some tweaking. It's working for me, but it's a bit messed up.

**Comment by csell5** [comment]
yea.. we should just shrink the font on smaller devices

**Comment by csell5** [comment]
and the search box, has and is janky on mobile too.

**Comment by asharonbaltazar** [comment]
Yeah, it doesn't work for me either. I'm not near a computer now, but is it possible my changes somehow got messed up with all the commits and rebasing? 

What's the group's slack? Maybe we should take these conversations there.

**Comment by csell5** [comment]
could be, not sure to be honest.

thatslack.thatconference.com

**Pull request merged**